libraries/botbuilder-core/src/skypeMentionNormalizeMiddleware.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Activity, Middleware, TurnContext } from 'botbuilder-core';
+
+
+/**
+ * Middleware to patch mention Entities from Skype since they don't conform to expected values.
+ * Bots that interact with Skype should use this middleware if mentions are used.
+ *
+ * @remarks
+ * A Skype mention "text" field is of the format:
+ *   <at id="28:2bc5b54d-5d48-4ff1-bd25-03dcbb5ce918">botname</at>
+ * But Activity.Text doesn't contain those tags and RemoveMentionText can't remove
+ * the entity from Activity.Text.
+ * This will remove the <at> nodes, leaving just the name.
+ */
+export class SkypeMentionNormalizeMiddleware implements Middleware {
+    public static normalizeSkypeMentionText(activity: Activity){
```

Comment:
I don't think this code will pass the eslint rules. 

Can you run this through the linter and cleanup the issues it brings up?

Reply:
@jspruance, this method blows up if there is no activity.text: message activities can be sent to the bot that have an attachment and no text. Can you make this more resilient and add more tests?

Reply:
this is done